src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -1,462 +1,445 @@
+            Assert.All(paths, (path) =>
+            {
+                Assert.Throws<PathTooLongException>(() => Create(path));
+            });
         }
-    }
 
-    [Fact]
-    public static void CreateDirectory_ExistingDirectoryWithTrailingSlashAsPath_DoesNotThrow()
-    {
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        [ActiveIssue(645)]
+        public void UnixPathLongerThan256_Allowed()
         {
-            string path = IOServices.AddTrailingSlashIfNeeded(directory.Path);
-
-            DirectoryInfo result = Directory.CreateDirectory(path);
+            DirectoryInfo testDir = Create(GetTestFilePath());
+            PathInfo path = IOServices.GetPath(testDir.FullName, 257, IOInputs.MaxComponent);
+            Assert.All(path.SubPaths, (subpath) =>
+            {
+                DirectoryInfo result = Create(subpath);
 
-            Assert.Equal(path, result.FullName);
-            Assert.True(Directory.Exists(result.FullName));
+                Assert.Equal(subpath, result.FullName);
+                Assert.True(Directory.Exists(result.FullName));
+            });
         }
-    }
-
-    [Fact]
-    public static void CreateDirectory_DotWithoutTrailingSlashAsPath_DoesNotThrow()
-    {
-        DirectoryInfo result = Directory.CreateDirectory(Path.Combine(TestInfo.CurrentDirectory, ".")); // "Current" directory
-
-        Assert.True(Directory.Exists(result.FullName));
-    }
 
-    [Fact]
-    public static void CreateDirectory_DotWithTrailingSlashAsPath_DoesNotThrow()
-    {
-        DirectoryInfo result = Directory.CreateDirectory(Path.Combine(TestInfo.CurrentDirectory, ".") + Path.DirectorySeparatorChar); // "Current" directory
-
-        Assert.True(Directory.Exists(result.FullName));
-    }
-
-    [Fact]
-    public static void CreateDirectory_DotDotWithoutTrailingSlashAsPath_DoesNotThrow()
-    {
-        DirectoryInfo result = Directory.CreateDirectory(Path.Combine(TestInfo.CurrentDirectory, ".."));    // "Parent" of current directory
-
-        Assert.True(Directory.Exists(result.FullName));
-    }
-
-    [Fact]
-    public static void CreateDirectory_DotDotWithTrailingSlashAsPath_DoesNotThrow()
-    {
-        DirectoryInfo result = Directory.CreateDirectory(Path.Combine(TestInfo.CurrentDirectory, "..") + Path.DirectorySeparatorChar);    // "Parent" of current directory
-
-        Assert.True(Directory.Exists(result.FullName));
-    }
-
-#if !TEST_WINRT // Cannot set current directory to root from appcontainer with it's default ACL
-    /*
-    [Fact]
-    [ActiveIssue(1220)] // SetCurrentDirectory
-    public static void CreateDirectory_DotDotAsPath_WhenCurrentDirectoryIsRoot_DoesNotThrow()
-    {
-        string root = Path.GetPathRoot(Directory.GetCurrentDirectory());
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void UnixPathLongerThan256_Throws()
```

Comment:
It's a bad method that works because of the values of MaxComponent and MaxPath. Subpaths will only have one element that is a fine test against MaxPath, but the iteration of that array is incorrect in the general case. It should look like:

```
DirectoryInfo testDir = Create(GetTestFilePath());
PathInfo path = IOServices.GetPath(testDir.FullName, 257, IOInputs.MaxComponent);
Assert.Throws<PathTooLongException>(() => Create(path.FullPath));
```